### PR TITLE
Fix HTTPUpdate flash size check and add SPIFFS size check

### PR DESCRIPTION
**Pull request by Jeroen88**
HTTPUpdate checked if the new sketch would fit into the _current_ partition free space, which was wrong. Now it checks if it fits into the _next_ flash partition.

A size check for a SPIFFS update is added, this was stil 'to do' from the previous PR.

Still to do: the SHA256 hash could be checked if a server sends it in the request header.

This PR includes PR #2155.

### CODE REVIEW

File: libraries/HTTPUpdate/src/HTTPUpdate.cpp [hunk 1]
```diff
@@ -366,6 +379,10 @@ bool HTTPUpdate::runUpdate(Stream& in, uint32_t size, String md5, int command)
         }
     }
 
+// To do: the SHA256 could be checked if the server sends it
+
+    delay(0);
```
Comment (me-no-dev) [inline libraries/HTTPUpdate/src/HTTPUpdate.cpp]: why?

File: libraries/HTTPUpdate/src/HTTPUpdate.cpp [hunk 2]
```diff
@@ -229,14 +231,25 @@ HTTPUpdateResult HTTPUpdate::handleUpdate(HTTPClient& http, const String& curren
...
+                }
+
+                if(len > _partition->size) {
+                    log_e("spiffsSize to low (%d) needed: %d\n", _partition->size, len);
+                    startUpdate = false;
+                }
             } else {
-                if(len > (int) ESP.getFreeSketchSpace()) {
-                    log_e("FreeSketchSpace to low (%d) needed: %d\n", ESP.getFreeSketchSpace(), len);
+                const esp_partition_t* _partition = esp_ota_get_next_update_partition(NULL);
```
Comment (me-no-dev) [inline libraries/HTTPUpdate/src/HTTPUpdate.cpp]: please move this logic to Esp.cpp and replace the one in `ESP.getFreeSketchSpace`. What is currently there makes no sense for any use.

File: libraries/HTTPUpdate/src/HTTPUpdate.cpp (hunk 1 above)
Comment (Jeroen88) [reply libraries/HTTPUpdate/src/HTTPUpdate.cpp]: Removing...

**Comment by me-no-dev** [comment]
Nice :) thanks!

**Comment by Jeroen88** [comment]
You're welcome! :)

**Pull request merged**